build-system/tasks/karma.conf.js

```diff
@@ -69,6 +78,19 @@ module.exports = {
     transform: [['babelify', {caller: {name: 'test'}, global: true}]],
     // Prevent "cannot find module" errors on Travis. See #14166.
     bundleDelay: isTravisBuild() ? 5000 : 1200,
+
+    persistentCache: browserifyPersistFs(
+      './build/.karma-cache',
```

Comment:
> Isn't this desirable?

This is not desirable because the zipped build output is used in other places (e.g. PR Deploy), and we don't want to include a large testing-only cache directory.

The alternative you've implemented (Travis caching) gives us what we need though.

Reply:
@jridgewell I'm unresolving this thread because I don't think the `browserify` cache actually works on Travis. For example, see this [build](https://travis-ci.org/github/ampproject/amphtml/jobs/685815303) on `master`:

- `.karma-cache` [is being copied](https://travis-ci.org/github/ampproject/amphtml/jobs/685815303#L268-L296) from the Travis cache (expand _"Setting up build cache"_)
- Yet, transforms are taking [> 1 minute](https://travis-ci.org/github/ampproject/amphtml/jobs/685815303#L1921-L1924) instead of a few seconds
- `.karma-cache` [is being updated](https://travis-ci.org/github/ampproject/amphtml/jobs/685815303#L1980-L1992) for the files in the PR (expand _"store build cache"_)

This appears to be happening for every single build. Perhaps we're busting the cache too aggressively?

(If this can be fixed, we can get a ~5 minute performance boost for Travis builds overall.)

Reply:
We merge package updates 2000000 times a day, which busts the cache. Here's a [build](https://travis-ci.org/github/ampproject/amphtml/builds/685792545) that did not have a package update directly before it, and it [browserified in 10s](https://travis-ci.org/github/ampproject/amphtml/jobs/685792555#L1933-L1936).

I chose this aggressive cache busting because it's difficult to determine what could affect the builds. Partly, that's because the browserify cache has to know what can change in browserify, and what can change in babelify.

We can fix this easily by merging package changes at overnight, and that'll keep the easy correctness guarantees.

Reply:
Gotcha, thanks for the context. I agree there's nothing wrong with the cache busting strategy.

Also, fair point re: package updates. I agree, we have too many. I've filed #28322 to discuss further.